src/ppl/src/IOPlacer.cpp

```diff
@@ -1931,11 +1976,18 @@ void IOPlacer::runAnnealing(bool random)
   initMirroredPins(true);
   initConstraints(true);
 
-  ppl::SimulatedAnnealing annealing(
+  annealing_ = new ppl::SimulatedAnnealing(
       netlist_io_pins_.get(), core_.get(), slots_, constraints_, logger_, db_);
```

Comment:
Why does this need to be a member rather than local?

Reply:
When the following method is called, SimulatedAnnealing object takes ownership of the renderer.
https://github.com/The-OpenROAD-Project/OpenROAD/blob/814b7ea72d6b7fdd8f1dea2ac4ba149c97ec85b5/src/ppl/src/SimulatedAnnealing.cpp#L167-L171
If the SimulatedAnnealing object was to remain local, then the renderer would run out of scope when `runAnnealing()` ends and for the renderer to work it needs to exist as long as the IOPlacer exists.

Reply:
Why - if there is nothing left to render then it can be deregistered and destroyed.

Reply:
I had misunderstood the way the renderer works. Indeed, it will be possible to see the annealing iterations if the object is local.
It only makes sense to make the object a member if it's desired to keep the last state in the layoutviewer after the annealing run is over (this way it will be possible to see the fly wires of the solution and the correct placement of the markers). Should I keep it local?

Reply:
IMO, we don't need to keep the final state after completing the annealing. We can just select the pins in their final positions to visualize the fly wires.

Reply:
Does the renderer store enough state to draw even after the annealer is destroyed?  If so then I would keep it as it but not pass ownership to the annealer.

Reply:
The only advantage I can see to keeping afterwards it is if it grows to show more data than the fly lines (eg constraints)

Reply:
The renderer stores the state in a vector of IOPins and a vector of vectors of InstancePins.

How about keep it local for now and later, if desired, change to show more data?

Reply:
fine